spotbugsTestCases/src/java14/IncorrectSelfComparasionInstanceOfPatternMatching.java

```diff
@@ -0,0 +1,15 @@
+/**
+ * False positive SA_LOCAL_SELF_COMPARISON reported on number when using instanceof pattern matching in Java 14.
+ * 
+ * @author Ken Schosinsky
+ */
+public class IncorrectSelfComparasionInstanceOfPatternMatching {
+
+    public static void instanceofPatternMatching() {
+        Number number = 1f;
+        if (number instanceof Float f) {

```

Comment:
Is this correct syntax? Why the `f`?

Reply:
Yes, this is the new syntax introducted in this preview feature. I suppose the false positive is triggered because number and f reference the same instance of the boxed `1f`.